### DISCUSSION TITLE
Should this add-on provide outbound connectivity - e.g. for SMB media

### DISCUSSION THREAD

**Discussion by a8ree** [hassio-addons/app-tailscale#236] [General]
I'd like to connect my HA instance to a cloud VM that has some storage. When connected to my tailscale network I am unable to ping or connect via SMB to other devices. Is this not supported or an issue with my config?

**Comment by lmagyar** [comment]
The current last release uses userspace networking (there is no tailscale0 interface to route traffic outward), so this is not possible.

Not using userspace networking is merged, but is not released due to a blockig issue: if you advertise the same subnets (eg. 192.168.1.x/24) on multiple machines (and there is a good chance that people we make this mistake), the HA instance will lose all it's network access (gets soft bricked). It's a known issue with Tailscale for years, no solution.

I don't now what is the plan to solve this, but I forked this add-on and made a workaround for this (adding local advertised subnets back to the routing table with higher priority than Tailscale's), you can install and use/test it form here: https://github.com/lmagyar/homeassistant-addon-tailscale

There is a PR to merge this solution to this "official" Tailscale add-on, but it is still open: #201 

If you try my fork, add this to the config options:
```
userspace_networking: false
```

If you have any problems, please open an issue in my forked repo.

**Reply by a8ree** [reply]
Thanks @lmagyar 

I've installed your fork but not seeing any difference. I will ping you on your repo!

**Comment by oleksii-laa** [comment]
Hello. I'm trying to set up my Home Assistant backup to my VPS server using SMB share and Tailscale. It seems that the "Add network storage" dialog in HA accepts only SMB shares on the same subnet and doesn't allow Tailscale IP address.  Maybe I need to translate my VPS Tailscale IP into the local IP, then connect to a network storage using this local address. Could someone help to configure this?

**Reply by lmagyar** [reply]
Did you disable userspace networking in the TS add-on's config? USE THE YAML CONFIG EDITOR, the UI editor is completely broken. After disabling it, you will be able to access other tailscale devices on your tailnet. Read the TS add-on's docs on userspace networking.

Additional info: https://community.home-assistant.io/t/tailscale-add-on-and-samba-network-storage-on-vps-with-tailscale/827045

**Reply by oleksii-laa** [reply]
Thank you for your reply. I just succeeded in adding my remote SMB share from my VPS. I made several attempts to disable Userspace in the HA Tailscale add-on.  I have successfully switched off userspace mode in my Home Assistant Tailscale add-on. I read earlier that the GUI add-on configuration could be buggy; the "Userspace" switch didn't work. After some manipulations (reloading HA YAML config, restarting HA) I have the "Userspace" turned off. I also found that the corresponding setting in the Tailscale add-on YAML file is: "userspace_networking: false".
